substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -108,28 +109,29 @@ private static int updateByteBuffer(int crc, long addr, int off, int len) {
 }
 
 @Platforms({Platform.LINUX.class, Platform.DARWIN.class})
-@TargetClass(java.util.zip.Deflater.class)
+@TargetClass(value = java.util.zip.Deflater.class)
 final class Target_java_util_zip_Deflater {
-
     @Alias //
-    @TargetElement(onlyWith = JDK8OrEarlier.class) //
+    @TargetElement(onlyWith = JDK10OrEarlier.class) //
     private byte[] buf;
-
     @Alias //
-    @TargetElement(onlyWith = JDK8OrEarlier.class) //
+    @TargetElement(onlyWith = JDK10OrEarlier.class) //
     int off;
-
     @Alias //
-    @TargetElement(onlyWith = JDK8OrEarlier.class) //
+    @TargetElement(onlyWith = JDK10OrEarlier.class) //
     int len;
+    @Alias //
+    private int level;
+    @Alias //
+    private int strategy;
+    @Alias //
+    private boolean setParams;
+    @Alias //
+    private boolean finish;
+    @Alias //
+    private boolean finished;
 
-    @Alias private int level;
-    @Alias private int strategy;
-    @Alias private boolean setParams;
-    @Alias private boolean finish;
-    @Alias private boolean finished;
-
-    // Deflater.c-Java_java_util_zip_Deflater_init(JNIEnv *env, jclass cls, jint level,
+    // (JDK 11)src/java.base/share/native/libzip/Deflater.c-Java_java_util_zip_Deflater_init(JNIEnv *env, jclass cls, jint level, jboolean nowrap)
     @Substitute
     private static long init(int level, int strategy, boolean nowrap) {
         z_stream strm = LibC.calloc(WordFactory.unsigned(1), SizeOf.unsigned(z_stream.class));
```

Comment:
[An informational note: If you were not in Posix code, you could use `UnmanagedMemory.calloc(UnsignedWord)` to do the same thing in a platform-independent way.]